portfolio/src/test/java/com/google/sps/SentimentTest.java

```diff
@@ -0,0 +1,295 @@
+
+  private static final String STRING_WITH_ENTITIES = "Entity One and Entity Two";
+  private static final String HAS_LINK_FOR_ENTITY_ONE = 
+    "<a target=\"_blank\" href=\"entity_one.com\">Entity One</a> and Entity Two";
+  private static final String HAS_LINK_FOR_ENTITY_TWO = 
+    "Entity One and <a target=\"_blank\" href=\"entity_two.com\">Entity Two</a>";
+  private static final String HAS_LINKS_FOR_BOTH_ENTITIES = 
+    "<a target=\"_blank\" href=\"entity_one.com\">Entity One</a>" +
+    " and <a target=\"_blank\" href=\"entity_two.com\">Entity Two</a>";
+  private static final String ENTITY_ONE_NAME = "Entity One";
+  private static final String ENTITY_ONE_LINK = "entity_one.com";
+  private static final String ENTITY_TWO_NAME = "Entity Two"; 
+  private static final String ENTITY_TWO_LINK = "entity_two.com";
+
+  private static Map<String, String> ENTITY_ONE_METADATA_MAP = new HashMap<String, String>();
+  private static Map<String, String> ENTITY_TWO_METADATA_MAP = new HashMap<String, String>();
+
+  private static final String CATEGORY_NAME_ONE = "Category One";
+  private static final String CATEGORY_NAME_TWO = "Category Two";
+
+  private static boolean isDocument(Document document) {
+    return true;
+  }
+  private static boolean isAnalyzeEntitiesRequest(
+    AnalyzeEntitiesRequest analyzeEntitiesRequest) {
+    return true;
+  }
+  private static boolean isClassifyTextRequest(
+    ClassifyTextRequest classifyTextRequest) {
+    return true;
+  }
+
+  @Before
+  public void setUp() {
+    mockLanguageServiceClient = mock(LanguageServiceClient.class);
+    mockSentimentResponse = mock(AnalyzeSentimentResponse.class);
+    mockSentiment = mock(Sentiment.class);
+
+    mockAnalyzeEntitiesResponse = mock(AnalyzeEntitiesResponse.class);
+    mockEntityOne = mock(Entity.class);
+    mockEntityTwo = mock(Entity.class);
+
+    mockClassifyTextResponse = mock(ClassifyTextResponse.class);
+    mockClassificationCategoryOne = mock(ClassificationCategory.class);
+    mockClassificationCategoryTwo = mock(ClassificationCategory.class);
+
+    when(mockLanguageServiceClient.analyzeSentiment(argThat(SentimentTest::isDocument)))
+      .thenReturn(mockSentimentResponse);
+    when(mockSentimentResponse.getDocumentSentiment())
+      .thenReturn(mockSentiment);
+
+    when(mockLanguageServiceClient
+      .analyzeEntities(argThat(SentimentTest::isAnalyzeEntitiesRequest)))
+      .thenReturn(mockAnalyzeEntitiesResponse);
+    when(mockEntityOne.getName()).thenReturn(ENTITY_ONE_NAME);
+    when(mockEntityTwo.getName()).thenReturn(ENTITY_TWO_NAME);
+
+    when(mockClassificationCategoryOne.getName()).thenReturn(CATEGORY_NAME_ONE);
+    when(mockClassificationCategoryTwo.getName()).thenReturn(CATEGORY_NAME_TWO);
+
+    ENTITY_ONE_METADATA_MAP.put("wikipedia_url", ENTITY_ONE_LINK);
+    ENTITY_TWO_METADATA_MAP.put("wikipedia_url", ENTITY_TWO_LINK);
+  }
+
+  @Test 
+  public void negativeSentiment() {
```

Comment:
could you try to incorporate this comment into the test name? then you can get ride of this comment

Reply:
here and below